src/io/flutter/preview/PreviewView.java

```diff
@@ -249,34 +250,45 @@ protected String getElementText(Object element) {
     tree.addMouseListener(new MouseAdapter() {
       @Override
       public void mouseClicked(MouseEvent e) {
-        if (e.getClickCount() > 0) {
+        if (e.getClickCount() > 1) {
           final TreePath selectionPath = tree.getSelectionPath();
           if (selectionPath != null) {
-            final DefaultMutableTreeNode node = (DefaultMutableTreeNode)selectionPath.getLastPathComponent();
-            final OutlineObject object = (OutlineObject)node.getUserObject();
-            final FlutterOutline outline = object.outline;
-            final int offset = outline.getDartElement() != null ? outline.getDartElement().getLocation().getOffset() : outline.getOffset();
-            if (currentFile != null) {
-              isTreeClicking = true;
-              try {
-                new OpenFileDescriptor(project, currentFile, offset).navigate(e.getClickCount() > 1);
-              }
-              finally {
-                isTreeClicking = false;
-              }
-            }
+            selectPath(selectionPath, true);
           }
         }
       }
     });
 
+    tree.addTreeSelectionListener(e -> {
+      final TreePath selectionPath = e.getNewLeadSelectionPath();
+      if (selectionPath != null) {
+        ApplicationManager.getApplication().invokeLater(() -> selectPath(selectionPath, false));
```

Comment:
Isn't listener already executed in the UI thread?
Why do we want `invokeLater`?

Reply:
I was seeing exceptions without this:

```
write-unsafe context! Model changes are allowed from write-safe contexts only. Please ensure you're using invokeLater/invokeAndWait with a correct modality state (not "any"). See TransactionGuard documentation for details.
```